core/src/serde/deserializers.rs

```diff
@@ -78,6 +78,8 @@ pub trait Deserializer<'de>: Sized {
 
     // === Compound types (consumer pattern) ===
 
+    // TODO(unknown members): Union unknown types are not well supported by
+    //                        the current impl.
```

Comment:
We just need to add some customizable unknown arm, and have some protocol setting to either ignore or do some protocol specific behavior